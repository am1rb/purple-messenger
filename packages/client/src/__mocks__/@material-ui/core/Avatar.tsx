import React from "react";

interface AvatarProps {
  src?: string;
}

function MuiAvatar({ src }: AvatarProps) {
  return <img data-testid="mock-mui-avatar" src={src} />;
}

export default MuiAvatar;
