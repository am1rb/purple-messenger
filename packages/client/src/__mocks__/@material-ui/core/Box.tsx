import React from "react";

export interface BoxProps {
  children: React.ReactNode;
}

function MuiBox({ children }: BoxProps) {
  return <div data-testid="mock-mui-box">{children}</div>;
}

export default MuiBox;
