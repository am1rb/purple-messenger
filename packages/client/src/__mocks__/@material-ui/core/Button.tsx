import React from "react";

export interface ButtonProps {
  onClick?: () => void;
  children: React.ReactNode;
}

function MuiButton({ onClick, children }: ButtonProps) {
  return (
    <div data-testid="mock-mui-button" onClick={onClick}>
      {children}
    </div>
  );
}

export default MuiButton;
