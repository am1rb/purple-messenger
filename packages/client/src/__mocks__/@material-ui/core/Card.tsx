import React from "react";

export interface CardProps {
  children: React.ReactNode;
}

function MuiCard({ children }: CardProps) {
  return <div data-testid="mock-mui-card">{children}</div>;
}

export default MuiCard;
