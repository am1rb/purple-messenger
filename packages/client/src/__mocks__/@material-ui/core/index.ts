export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Card } from "./Card";
export { default as CardContent } from "./CardContent";
export { default as Typography } from "./Typography";
export { default as Button } from "./Button";
