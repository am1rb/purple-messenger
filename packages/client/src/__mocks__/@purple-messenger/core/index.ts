import * as purpleMessengerCore from "@purple-messenger/core";

const mockedPurpleMessengerCore = jest.genMockFromModule<
  typeof purpleMessengerCore
>("@purple-messenger/core");

function mockedAction(action = () => ({ type: "" })) {
  return jest.fn(action);
}

module.exports = {
  ...mockedPurpleMessengerCore,
  loadConversationList: mockedAction(),
  unloadConversationList: mockedAction(),
  signOut: mockedAction(),
  clearMessageList: mockedAction(),
  seenMessageAck: mockedAction(),
  start: mockedAction(),
  shutdown: mockedAction(),
  submitMessage: mockedAction(),
  startTypingMessage: mockedAction(),
  stopTypingMessage: mockedAction(),
  signIn: mockedAction(),
  setAuthError: mockedAction(),
  verifyToken: mockedAction(),
  connected: mockedAction(purpleMessengerCore.connected),
  disconnected: mockedAction(purpleMessengerCore.disconnected),
};
