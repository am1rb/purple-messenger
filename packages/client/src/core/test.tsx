import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { ConnectedRouter } from "connected-react-router";
import configureStore, { history } from "core/redux/configureStore";
import { State as RootState } from "core/redux/reducers";
import theme from "core/style/theme";
import { ThemeProvider } from "@material-ui/styles";

export function renderWithStore(
  ui: React.ReactElement,
  initStore: Partial<RootState>
) {
  const store = configureStore(initStore);

  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <ConnectedRouter history={history}>{children}</ConnectedRouter>
        </Provider>
      </ThemeProvider>
    ),
  });
}
