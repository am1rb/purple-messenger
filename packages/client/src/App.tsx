import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import PageRouter from 'components/PageRouter';
import {start, shutdown} from '@purple-messenger/core';
import './App.css';

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(start());
    return function() {
      dispatch(shutdown());
    }
  }, [ dispatch ]);
  return <PageRouter />;
};

export default App;
