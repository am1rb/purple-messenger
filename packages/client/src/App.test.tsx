import React from "react";
import App from "./App";
import { renderWithStore } from "core/test";

jest.mock("components/PageRouter");

it("renders without crashing", () => {
  const { container } = renderWithStore(<App />, {});
  expect(container).toMatchSnapshot();
});
