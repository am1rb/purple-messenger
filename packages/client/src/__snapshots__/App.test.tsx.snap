// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    data-testid="mock-page-router"
  />
</div>
`;
